### PR TITLE
Suggested tweaks to the supported features docs

### PR DESCRIPTION
**Pull request by shonfeder**
Some clarifications and slight rewording on the documentation of supported features.

There are also two places where I am not sure what is meant, I'll call those out with comments.

### CODE REVIEW

File: docs/src/apalache/features.md [hunk 1]
```diff
@@ -2,24 +2,24 @@ Here is the list of the TLA+ language features that are currently supported by A
...
 
 ## Language
 
 ### Module-Level constructs
 
 Construct  | Supported? | Milestone | Comment
 ------------------------|:------------------:|:---------------:|--------------
-``EXTENDS module`` | ✔ | - | As soon as SANY imports the module. Some standard modules are not supported yet
-``CONSTANTS C1, C2`` | ✔ | -  | Either define a ``ConstInit`` operator to initialize the constants, or declare operators instead of constants, e.g., C1 == 111
+``EXTENDS module`` | ✔ | - | As soon as SANY imports the module. <!-- What does the prev sentence mean? --> Some standard modules are not supported yet
```
Comment (shonfeder) [inline docs/src/apalache/features.md:17]: I don't understand what is meant in this context by the sentence 

> As soon as SANY imports the module.

File: docs/src/apalache/features.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ Operator  | Supported? | Milestone | Comment
 `f[e]` | ✔ | - |
 `DOMAIN f` | ✔ | - |
 `[ x \in S ↦ e]` | ✔ | - |
-`[ S -> T ]` | ✔ | - | Sometimes, the functions sets are expanded
+`[ S -> T ]` | ✔ | - | Sometimes, the functions sets are expanded <!-- What does this mean? -->
```
Comment (shonfeder) [inline docs/src/apalache/features.md:65]: I don't know what this comment is supposed to mean or what the impact is on users.

File: docs/src/apalache/features.md (hunk 1 above)
Comment (konnov) [reply docs/src/apalache/features.md:17]: I also don't understand that. It is all modulo SANY parsing things :)

Comment (konnov) [reply docs/src/apalache/features.md:17]: Let's remove that sentence

File: docs/src/apalache/features.md (hunk 2 above)
Comment (konnov) [reply docs/src/apalache/features.md:65]: It is confusing. It means that in some cases functions sets are treated purely symbolically, but sometimes the model checker would try to construct an explicit set of functions, which is exponential. Actually, the model checker would just fail there and will not try to construct such a set. So this sentence is not helping the user.

File: docs/src/apalache/features.md [hunk 3]
```diff
@@ -2,24 +2,24 @@ Here is the list of the TLA+ language features that are currently supported by A
...
 
-At the moment, Apalache is able to check invariants and inductive invariants.
-It means that you can only check safety with Apalache, unless you employ
-[liveness-to-safety] transformation in your spec. It does not support
-liveness properties.  If you really like to see liveness implemented, upvote
-the [liveness feature].
+At the moment, Apalache is only able to check invariants and inductive
+invariants.  Which means that you can only check safety properties with
+Apalache, unless you employ a [liveness-to-safety] transformation in your spec.
+In general, we doe not support checking liveness properties.  If you would like
```
Comment (konnov) [inline docs/src/apalache/features.md:8]:
```suggestion
In general, we do not support checking liveness properties.  If you would like
```

File: docs/src/apalache/features.md [hunk 4]
```diff
@@ -85,10 +85,10 @@ Note that our type system distinguishes tuples from general functions.
 
 Operator  | Supported? | Milestone | Comment
 ------------------------|:------------------:|:---------------:|--------------
-`e[i]` | ✔ / ✖ | - | Provided that ``i`` is a constant expression
-`<< e1, ..., e_n >>` | ✔ | - | By default, a tuple is constructed. Use a [type annotation](types-and-annotations.md) to construct a sequence of proper type.
+`e[i]` | ✔ / ✖ | - | Provided that `i` is a constant expression
+`<< e1, ..., e_n >>` | ✔ | - | By default, a tuple is constructed. Use a [type annotation](types-and-annotations.md) to construct a sequence.
```
Comment (konnov) [inline docs/src/apalache/features.md:89]: This is not even true anymore. You would need a type annotation in the both cases.

File: docs/src/apalache/features.md [hunk 5]
```diff
@@ -105,20 +105,20 @@ Construct  | Supported? | Milestone | Comment
 ------------------------|:------------------:|:---------------:|--------------
 `IF p THEN e1 ELSE e2` | ✔ | - | Provided that both e1 and e2 have the same type
 `CASE p1 -> e1 [] ... [] p_n -> e_n [] OTHER -> e` | ✔ | - | See the comment above
-`CASE p1 -> e1 [] ... [] p_n -> e_n` | ✖ | - | Introduce the default arm with `OTHER`.
-``LET d1 == e1 ... d_n == e_n IN e`` | ✔ |  | All applications of `d1`, ..., `d_n` are replaced with the expressions `e1`, ... `e_n` respectively. LET-definitions without arguments are kept in place.
+`CASE p1 -> e1 [] ... [] p_n -> e_n` | ✖ | - | The default arm with `OTHER` is always required
```
Comment (konnov) [inline docs/src/apalache/features.md:108]: This is also not true. I have implemented the translation for `CASE` without `OTHER` a few weeks ago, thanks to the new type checker :P

Comment (konnov) [inline docs/src/apalache/features.md:108]:
```suggestion
`CASE p1 -> e1 [] ... [] p_n -> e_n` | ✔| - |
```

File: docs/src/apalache/features.md (hunk 4 above)
Comment (konnov) [inline docs/src/apalache/features.md:89]:
```suggestion
`<< e1, ..., e_n >>` | ✔ | - | Use a [type annotation](https://apalache.informal.systems/docs/tutorials/snowcat-tutorial.html) to distinguish between a tuple and a sequence.
```

File: docs/src/apalache/features.md (hunk 2 above)
Comment (konnov) [inline docs/src/apalache/features.md:65]:
```suggestion
`[ S -> T ]` | ✔ | - |
```

File: docs/src/apalache/features.md (hunk 1 above)
Comment (konnov) [inline docs/src/apalache/features.md:17]:
```suggestion
``EXTENDS module`` | ✔ | - | a few standard modules are not supported yet (Bags and Reals)
```

**Review by konnov — APPROVED** [review]
Looks good! A few things were out of date, so it's good you looked at this doc.

File: docs/src/apalache/features.md (hunk 4 above)
Comment (shonfeder) [reply docs/src/apalache/features.md:89]: I'll fix this. Thanks.

File: docs/src/apalache/features.md (hunk 2 above)
Comment (shonfeder) [reply docs/src/apalache/features.md:65]: Cool. That helps. How about

> Supported, provided the function can be interpreted symbolically

**Pull request merged**